docs/csharp/whats-new/csharp-14.md

```diff
@@ -0,0 +1,100 @@
+---
+title: What's new in C# 14
+description: Get an overview of the new features in C# 14. C# 14 ships with .NET 10.
+ms.date: 02/19/2025
+ms.topic: whats-new
+---
+# What's new in C# 14
+
+C# 14 includes the following new features. You can try these features using the latest [Visual Studio 2022](https://visualstudio.microsoft.com/vs/preview/) version or the [.NET 10 SDK](https://dotnet.microsoft.com/download/dotnet):
+
+- [`nameof` supports unbound generic types](#unbound-generic-types-and-nameof)
+- [More implicit conversions for `Span<T>` and `ReadOnlySpan<T>`](#implicit-span-conversions)
+- [Modifiers on simple lambda parameters](#simple-lambda-parameters-with-modifiers)
+- [`field` backed properties](#the-field-keyword)
+
+C# 14 is supported on **.NET 10**. For more information, see [C# language versioning](../language-reference/configure-language-version.md).
+
+You can download the latest .NET 10 SDK from the [.NET downloads page](https://dotnet.microsoft.com/download). You can also download [Visual Studio 2022](https://visualstudio.microsoft.com/vs/), which includes the .NET 10 SDK.
+
+New features are added to the "What's new in C#" page when they're available in public preview releases. The [working set](https://github.com/dotnet/roslyn/blob/main/docs/Language%20Feature%20Status.md#working-set) section of the [roslyn feature status page](https://github.com/dotnet/roslyn/blob/main/docs/Language%20Feature%20Status.md) tracks when upcoming features are merged into the main branch.
+
+You can find any breaking changes introduced in C# 14 in our article on [breaking changes](~/_roslyn/docs/compilers/CSharp/Compiler%20Breaking%20Changes%20-%20DotNet%2010.md).
+
+[!INCLUDE [released-version-feedback](./includes/released-feedback.md)]
+
+## The `field` keyword
+
+The token `field` enables you to write a property accessor body without declaring an explicit backing field. The token `field` is replaced with a compiler synthesized backing field.
+
+For example, previously, if you wanted to ensure that a `string` property couldn't be set to `null`, you had to declare a backing field and implement both accessors:
+
+```csharp
+private string _msg;
+public string Message
+{
+    get => _msg;
+    set
+    {
+        if (value is null) throw new NullArgumentException(nameof(value));
+        _msg = value;
+    }
+}
+```
+
+You can now simplify your code to:
+
+```csharp
+public string Message
+{
+    get;
+    set
+    {
+        field = (value is not null) ? value : throw new NullArgumentException(nameof(value));
```

Comment:
How about using the `??` syntax here?

```suggestion
        field = value ?? throw new ArgumentNullException(nameof(value));
```

You could use it in the before example too. Both examples could then have expression bodies (i.e. `set => field ?? ...`).

Also, the exception type is `ArgumentNullException`, not `NullArgumentException`.